src/backend/expungeservice/form_filling.py

```diff
@@ -256,23 +260,34 @@ def _build_pdf_for_eligible_case(
             "dismissed_charges": "; ".join(dismissed_names),
             "dismissed_dates": "; ".join(dismissed_dates),
         }
-        form = from_dict(data_class=FormDataWithOrder, data=form_data_dict)
         location = case.summary.location.lower()
         pdf_path = FormFilling._build_pdf_path(location, convictions)
         base_file_name = FormFilling._build_base_file_name(location, convictions)
         file_name = os.path.basename(base_file_name)
         pdf = PdfReader(pdf_path)
-        for field in pdf.Root.AcroForm.Fields:
-            field_name = field.T.lower().replace(" ", "_").replace("(", "").replace(")", "")
-            field_value = getattr(form, field_name)
-            field.V = field_value
-            warnings += FormFilling._set_font(field, field_value)
-        for page in pdf.pages:
-            annotations = page.get("/Annots")
-            if annotations:
-                for annotation in annotations:
-                    annotation.update(PdfDict(AP=""))
-        pdf.Root.AcroForm.update(PdfDict(NeedAppearances=PdfObject("true")))
+
+        if "oregon" in pdf_path:
+            new_pdf = PDF(pdf_path, {"full_path": True})
+            new_pdf.update_annotations(form_data_dict)
+            warnings = new_pdf.warnings
+            pdf = new_pdf._pdf
+        else:
+            for field in pdf.Root.AcroForm.Fields:
+                form = from_dict(data_class=FormDataWithOrder, data=form_data_dict)
+                field_name = field.T.lower().replace(" ", "_").replace("(", "").replace(")", "")
+                field_value = getattr(form, field_name)
+                field.V = field_value
+                warnings += FormFilling._set_font(field, field.V)
+
+            # Since we are setting the values of the AcroForm.Fields, we need to
+            # remove the Appearance Dictionary ("/AP") of the PDF annotations in
+            # order for the value to appear in the PDF.
+            for page in pdf.pages:
+                annotations = page.get("/Annots")
+                if annotations:
+                    for annotation in annotations:
+                        annotation.update(PdfDict(AP=""))
+            pdf.Root.AcroForm.update(PdfDict(NeedAppearances=PdfObject("true")))
```

Comment:
If I had to nitpick, there is two different levels of abstraction going on here. Most of the else could probably use something similar to your `update_annotations`. Just for the future. I don't want you to do another testing cycle here.

Reply:
Appreciate the feedback and agree. I hadn't looked at the Multnomah county forms until the last minute yesterday. I'm planning to unify the code more and hopefully I can just use the annotations approach with the Multnomah forms.